drivers/sensor/bme680/bme680.c

```diff
@@ -0,0 +1,426 @@
+	if (err < 0) {
+		return err;
+	}
+
+	/* Temperature related coefficients */
+	data->par_t1 = (u16_t)(BME680_CONCAT_BYTES(buff[32], buff[31]));
+	data->par_t2 = (s16_t)(BME680_CONCAT_BYTES(buff[1], buff[0]));
+	data->par_t3 = (u8_t)(buff[2]);
+
+	/* Pressure related coefficients */
+	data->par_p1 = (u16_t)(BME680_CONCAT_BYTES(buff[5], buff[4]));
+	data->par_p2 = (s16_t)(BME680_CONCAT_BYTES(buff[7], buff[6]));
+	data->par_p3 = (s8_t)buff[8];
+	data->par_p4 = (s16_t)(BME680_CONCAT_BYTES(buff[11], buff[10]));
+	data->par_p5 = (s16_t)(BME680_CONCAT_BYTES(buff[13], buff[12]));
+	data->par_p6 = (s8_t)(buff[15]);
+	data->par_p7 = (s8_t)(buff[14]);
+	data->par_p8 = (s16_t)(BME680_CONCAT_BYTES(buff[19], buff[18]));
+	data->par_p9 = (s16_t)(BME680_CONCAT_BYTES(buff[21], buff[20]));
+	data->par_p10 = (u8_t)(buff[22]);
+
+	/* Humidity related coefficients */
+	data->par_h1 = (u16_t)(((u16_t)buff[25] << 4) | (buff[24] & 0x0f));
+	data->par_h2 = (u16_t)(((u16_t)buff[23] << 4) | ((buff[24]) >> 4));
+	data->par_h3 = (s8_t)buff[26];
+	data->par_h4 = (s8_t)buff[27];
+	data->par_h5 = (s8_t)buff[28];
+	data->par_h6 = (u8_t)buff[29];
+	data->par_h7 = (s8_t)buff[30];
+
+	/* Gas heater related coefficients */
+	data->par_gh1 = (s8_t)buff[35];
+	data->par_gh2 = (s16_t)(BME680_CONCAT_BYTES(buff[34], buff[33]));
+	data->par_gh3 = (s8_t)buff[36];
+
+	data->res_heat_val = (s8_t)buff[37];
+	data->res_heat_range = ((buff[39] & BME680_MSK_RH_RANGE) >> 4);
+	data->range_sw_err = ((s8_t)(buff[41] & BME680_MSK_RANGE_SW_ERR)) / 16;
+
+	return 0;
+}
+
+static int bme680_chip_init(struct device *dev)
+{
+	struct bme680_data *data = (struct bme680_data *)dev->driver_data;
+	int err;
+
+	err = bme680_reg_read(data, BME680_REG_CHIP_ID, &data->chip_id, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	if (data->chip_id == BME680_CHIP_ID) {
+		SYS_LOG_ERR("BME680 chip detected");
+	} else {
+		SYS_LOG_ERR("Bad BME680 chip id 0x%x", data->chip_id);
+		return -ENOTSUP;
+	}
+
+	err = bme680_read_compensation(data);
+	if (err < 0) {
+		return err;
+	}
+
+	err = bme680_reg_write(data, BME680_REG_CTRL_HUM, BME680_HUMIDITY_OVER);
+	if (err < 0) {
+		return err;
+	}
+
+	err = bme680_reg_write(data, BME680_REG_CONFIG, BME680_CONFIG_VAL);
+	if (err < 0) {
+		return err;
+	}
+
+	err = bme680_reg_write(data, BME680_REG_CTRL_GAS_1,
+			       BME680_CTRL_GAS_1_VAL);
+	if (err < 0) {
+		return err;
+	}
+
+	err = bme680_reg_write(data, BME680_REG_RES_HEAT0,
+			       bme680_calc_res_heat(data, BME680_HEATR_TEMP));
+	if (err < 0) {
+		return err;
+	}
+
+	err = bme680_reg_write(data, BME680_REG_GAS_WAIT0,
+			       bme680_calc_gas_wait(BME680_HEATR_DUR_MS));
+	if (err < 0) {
+		return err;
+	}
+
+	err = bme680_reg_write(data, BME680_REG_CTRL_MEAS,
+			       BME680_CTRL_MEAS_VAL);
+	if (err < 0) {
+		return err;
+	}
+
+	return 0;
+}
+
+static int bme680_init(struct device *dev)
+{
+	struct bme680_data *data = dev->driver_data;
+
+	data->i2c_master = device_get_binding(
+		CONFIG_BME680_I2C_NAME);
```

Comment:
There should be a const device config struct that you can read this from, rather than using the macro directly. See the fxos8700 sensor driver for an example. Same goes for the I2C name and address.